### ISSUE TITLE
TypeScript definition

### ISSUE THREAD

**Issue by nya1** [RuntimeTools/chai-openapi-response-validator#61]
Hello,
I would like to use this module in my typescript project but currently is missing the extended chai methods such as `satisfySchemaInApiSpec`, it would be nice to have the typescript definition

**Comment by rwalle61** [comment]
Thanks for raising this! I have wanted to support Typescript features for a while but not had time to implement it, especially as I’m fairly new to Typescript.

To clarify:
* are you unable to use this module at all in your Typescript project, because we are missing some type definitions? 
* What is the minimum change needed for this module to work in your project?
* what are the Typescript changes you would like to see to this module, after that first step?

**Comment by nya1** [comment]
> are you unable to use this module at all in your Typescript project, because we are missing some type definitions?

I'm able to use this module but I need some workarounds to not have type errors

> What is the minimum change needed for this module to work in your project? Is it to add some types like for the standard chai assertions https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/chai/v2/index.d.ts ?

> what are the Typescript changes you would like to see to this module, after that first step?

Yes I think is the only thing needed to make it work properly

Other examples are: 

- https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/chai-fuzzy
- https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/chai-uuid
- https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/chai-almost

**Comment by rwalle61** [comment]
Glad you can still run this module, albeit with some workarounds.

Thanks for the examples, sounds like a good change.

I'm currently fixing another bug (issue 54) but when I have time I will look into this. Happy to take a PR earlier if you fancy it

**Comment by rwalle61** [comment]
@nya1 would you like to pull that branch to check it solves your issue?

**Comment by nya1** [comment]
I'll give a try this weekend, it looks good, maybe I think the return type `Assertion` can be replaced with `void` because in your JS code you cannot chain other chai methods to satisfyApiSpec & satisfySchemaInApiSpec methods

**Comment by rwalle61** [comment]
Thanks!

Re chaining other chai methods, I seem to be able to - these are hitting the `foo` assertions
```js
expect(res).to.satisfyApiSpec.and.to.equal('foo');
expect(obj).to.satisfySchemaInApiSpec(schemaName).and.to.equal('foo');
```
(Not that I would recommend chaining like this)

**Comment by rwalle61** [comment]
Thanks for checking the Declaration File works. I've released it in 0.8.0 so feel free to close this when you're happy.

0.8.0 also makes the assertion failure messages much more informative, so any feedback (especially problems) with that will be appreciated!
